crates/wallet/src/wallet/error.rs

```diff
@@ -127,12 +127,16 @@ impl fmt::Display for CreateTxError {
                 requested,
                 required,
             } => {
-                write!(f, "TxBuilder requested timelock of `{:?}`, but at least `{:?}` is required to spend from this script", required, requested)
+                if requested.is_block_height() && required.is_block_height() {
+                    write!(f, "TxBuilder requested timelock of `{}` blocks, but at least `{}` blocks are required to spend from this script", required, requested)
+                } else {
+                    write!(f, "TxBuilder requested timelock of `{}` seconds, but at least `{}` seconds are required to spend from this script", required, requested)
+                }
```

Comment:
I think it's fine to display LockTime unit-less, but if we want the units you could use the alternate format syntax `{:#}`.
```suggestion
                write!(f, "TxBuilder requested timelock of `{}`, but at least `{}` is required to spend from this script", required, requested)

```